### ISSUE TITLE
[a11y]: Request to backport input and role fix for `SelectableTile` component from PR #12031 to v10

### ISSUE THREAD

**Issue by kenneth-nguyen1** [carbon-design-system/carbon#14321]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.59.5

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Description

The SelectableTile component has a critical a11y violation
"The tabbable element's role 'none' is not a widget role"
using the IBM Accessibility Checker tool.

I saw the issue was fixed in [PR 12031](https://github.com/carbon-design-system/carbon/pull/12301) for v11, and now requesting for the fix to also be backported to v10. That would be much appreciated!

The violation is reproducible on the [v10](https://v7-react.carbondesignsystem.com/?path=/story/components-tile--multi-select) storybook.
The violation is not reproducible on the [v11](https://react.carbondesignsystem.com/?path=/story/components-tile--multi-select) storybook.

![Screenshot 2023-07-26 at 2 06 51 PM](https://github.com/carbon-design-system/carbon/assets/114688923/6f5f844c-522e-4bc2-a3ec-4bf1ad6cb7b8)

### Reproduction/example

Storybook links for v10 and v11 in description above

### Steps to reproduce

On the storybook link (either v10 or v11) for the Tile -> Multi Select story, run the IBM Accessibility Checker scan. The scan should reports the violation for v10 but not for v11.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@kenneth-nguyen1 Hey, sorry we missed your comment on the original PR for v11. We'll take a look at backporting this in our next sprint, unless you'd like to work on a PR?

**Comment by kenneth-nguyen1** [comment]
@tay1orjones No worries, and yeah I would like to try to work on a PR to backport this fix!

This would be my first contribution here, so I will comment again in the next day or so if I run into any issues :)

**Comment by tay1orjones** [comment]
Sounds good @kenneth-nguyen1, once your PR is up and merged it'll be @andreancardona and @alisonjoseph who would cut a new patch version for v10 including the fix.
